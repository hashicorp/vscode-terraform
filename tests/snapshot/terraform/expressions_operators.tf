# arithmetic

1 + 2 * 3 / 4

5 % 6

-thing

# equality

thing1 == thing2
thing1 != thing2

# comparison

thing1 > thing2
thing1 >= thing2
thing1 < thing2
thing1 <= thing2

# logical

thing1 || thing2
thing1 && thing2
!thing1
