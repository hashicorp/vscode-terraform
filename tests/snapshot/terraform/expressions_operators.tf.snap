># arithmetic
#^ scope.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^ scope.terraform comment.line.terraform
>
>1 + 2 * 3 / 4
#^ scope.terraform constant.numeric.integer.terraform
# ^ scope.terraform
#  ^ scope.terraform keyword.operator.arithmetic.terraform
#   ^ scope.terraform
#    ^ scope.terraform constant.numeric.integer.terraform
#     ^ scope.terraform
#      ^ scope.terraform keyword.operator.arithmetic.terraform
#       ^ scope.terraform
#        ^ scope.terraform constant.numeric.integer.terraform
#         ^ scope.terraform
#          ^ scope.terraform keyword.operator.arithmetic.terraform
#           ^ scope.terraform
#            ^ scope.terraform constant.numeric.integer.terraform
>
>5 % 6
#^ scope.terraform constant.numeric.integer.terraform
# ^ scope.terraform
#  ^ scope.terraform keyword.operator.arithmetic.terraform
#   ^ scope.terraform
#    ^ scope.terraform constant.numeric.integer.terraform
>
>-thing
#^ scope.terraform keyword.operator.arithmetic.terraform
# ^^^^^^ scope.terraform
>
># equality
#^ scope.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^ scope.terraform comment.line.terraform
>
>thing1 == thing2
#^^^^^^^ scope.terraform
#       ^^ scope.terraform keyword.operator.terraform
#         ^^^^^^^^ scope.terraform
>thing1 != thing2
#^^^^^^^ scope.terraform
#       ^^ scope.terraform keyword.operator.terraform
#         ^^^^^^^^ scope.terraform
>
># comparison
#^ scope.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^ scope.terraform comment.line.terraform
>
>thing1 > thing2
#^^^^^^^ scope.terraform
#       ^ scope.terraform keyword.operator.terraform
#        ^^^^^^^^ scope.terraform
>thing1 >= thing2
#^^^^^^^ scope.terraform
#       ^^ scope.terraform keyword.operator.terraform
#         ^^^^^^^^ scope.terraform
>thing1 < thing2
#^^^^^^^ scope.terraform
#       ^ scope.terraform keyword.operator.terraform
#        ^^^^^^^^ scope.terraform
>thing1 <= thing2
#^^^^^^^ scope.terraform
#       ^^ scope.terraform keyword.operator.terraform
#         ^^^^^^^^ scope.terraform
>
># logical
#^ scope.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^ scope.terraform comment.line.terraform
>
>thing1 || thing2
#^^^^^^^ scope.terraform
#       ^^ scope.terraform keyword.operator.logical.terraform
#         ^^^^^^^^ scope.terraform
>thing1 && thing2
#^^^^^^^ scope.terraform
#       ^^ scope.terraform keyword.operator.logical.terraform
#         ^^^^^^^^ scope.terraform
>!thing1
#^ scope.terraform keyword.operator.logical.terraform
# ^^^^^^^ scope.terraform
>